Cython/Compiler/ModuleNode.py

```diff
@@ -3082,6 +3082,14 @@ def mod_init_func_cname(self, prefix, env):
         # from PEP483
         return self.punycode_module_name(prefix, env.module_name)
 
+    # Returns the name of the C-function that corresponds to the module initialisation.
+    # (module initialisation == the cython code outside of functions)
+    # Note that this should never be the name of a wrapper and always the name of the
+    # function containing the actual code. Otherwise, cygdb will experience problems.
+    def module_init_func_cname(self):
```

Comment:
For consistency with the other methods, I would still pass `env` as argument here.
Also, this is the "module exec" function, not the "module init" function (since PEP-489).
In fact, the name of the function (exec or init) that populates the module namespace depends on whether PEP-489 is used or not, which is a C compile time setting.
I guess we could unify that by always creating the exec function and passing it the module object in the non-PEP489 case, but that would be a separate PR.
In any case, PEP-489 is the default way to init a module now.

Reply:
We call module_init_func_cname in ParseTreeTransforms.py. There is no `env` variable there, so we would have to use `node.module_init_func_cname(node.scope)`. This doesn't look good to me. I would suggest renaming the function to module_exec_func_cname and not changing anything else.
Maybe we should have tests for both the PEP-489 and the non-PEP-489 case.
What's the reason why Cython supports both using PEP-489 and not using it?

Reply:
> Maybe we should have tests for both the PEP-489 and the non-PEP-489 case.

In this case, yes. In general, I think it's ok if it comes implicitly with the Python version.

> What's the reason why Cython supports both using PEP-489 and not using it?

I guess we may remove the non-PEP-489 fallback at some point, but as long as Cython's support for it is not complete (including module reloading and subinterpreters), and as long as we support Python 2.7, it's better to give users a chance to disable this if it ever gets in the way. Second best answer: because we can. :)

> renaming the function to module_exec_func_cname and not changing anything else.

Except that it's currently more than one function name that you're interested in here.